File: 0705-design-hashset/0705-design-hashset.kt
class MyHashSet() {
    private val SIZE = 1000000
    private val buckets: Array<MutableList<Int>> = Array(SIZE) { mutableListOf<Int>() }

    fun add(key: Int) {
        val bucketIndex = getBucketIndex(key)
        if (!contains(key)) {
            buckets[bucketIndex].add(key)
        }
    }

    fun remove(key: Int) {
        val bucketIndex = getBucketIndex(key)
        buckets[bucketIndex].remove(key)
    }

    fun contains(key: Int): Boolean {
        val bucketIndex = getBucketIndex(key)
        return key in buckets[bucketIndex]
    }

    private fun getBucketIndex(key: Int): Int {
        return key % SIZE
    }
}


/**
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */

File: 0705-design-hashset/0705-design-hashset.cpp
class MyHashSet {
private:
    int MAX_SIZE = 1000001;
vector<bool> keySet;
public:
    MyHashSet() : keySet(MAX_SIZE) {
    }
    
    void add(int key) {
        keySet[key] = true;
    }
    
    void remove(int key) {
        keySet[key] = false;
    }
    
    bool contains(int key) {
        return keySet[key];
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

